### PR TITLE
[fea-rs] Stop sorting lookups in feature

### PR DESCRIPTION
**Pull request by cmyr**
The order of lookups in the feature doesn't matter, so this is not a functional change; prevoiusly we sorted because sorting is required for the 'dedup' method on Vec to do what we want.

The sorting was something we did that fonttools didn't, and I'd like to try and converge as much as possible.

### CODE REVIEW

**Comment by rsheeter** [comment]
I immediately wonder why normalization doesn't take care of this for us?

**Review by rsheeter — APPROVED** [review]

**Comment by anthrotype** [comment]
> I immediately wonder why normalization doesn't take care of this for us?

sure, otl-normalizer could/should take care of this as well, but when it doesn't cost much to converge with existing python implementation, there's no reason to diverge just for the sake of it.

**Review by anthrotype — APPROVED** [review]

File: fea-rs/src/compile/features.rs [hunk 1]
```diff
@@ -177,14 +177,13 @@ impl AllFeatures {
...
     }
 
-    pub(crate) fn sort_and_dedupe_lookups(&mut self) {
+    pub(crate) fn dedupe_lookups(&mut self) {
         // if any duplicate lookups have made their way into our features, remove them;
         // they will be ignored by the shaper anyway.
+        let mut seen = HashSet::new();
         for lookup in self.features.values_mut() {
-            // note that the order of lookups in a feature doesn't matter, they
-            // are processed in the order that they appear in the lookup list.
```
[the remark this answers is not in the document]
Comment (anthrotype) [reply fea-rs/src/compile/features.rs:185]: the comment was there to justify the sorting behavior. IMO it doesn't need to justify why it is _not_ sorting. It's not that one or ther other is better/more correct.

**Comment by cmyr** [comment]
> > I immediately wonder why normalization doesn't take care of this for us?
> 
> sure, otl-normalizer could/should take care of this as well, but when it doesn't cost much to converge with existing python implementation, there's no reason to diverge just for the sake of it.

cosimo is correct here, `otl-normalizer` _does_ handle this but the divergence is unnecessary, and it isn't clear we want to use otl-normalizer on GSUB.

**Pull request merged**